src/script/script.cpp

```diff
@@ -255,6 +256,93 @@ bool CScript::IsPayToScriptHash() const
             (*this)[22] == OP_EQUAL);
 }
 
+bool CScript::IsRedeemScriptReveal(CScript scriptpubkey) const{
+    CScript check_spk, redeemScript = scriptpubkey;
+    if (
+        // these magic numbers correspond to a typical(as of May 2021) atomicdex HLTC
+        redeemScript.size() == 110 &&
+        redeemScript[0] == OP_RETURN &&
+        redeemScript[3] == OP_IF &&
+        redeemScript[9] == OP_NOP2 &&
+        redeemScript[10] == OP_DROP &&
+        redeemScript[45] == OP_CHECKSIG &&
+        redeemScript[46] == OP_ELSE &&
+        redeemScript[47] == OP_SIZE &&
+        redeemScript[48] == 0x01 &&
+        redeemScript[49] == 0x20 &&
```

Comment:
```
redeemScript[49] == 0x20
```
@artemii235 May be i'm wrong, but isn't it should be **0x14 (20 bytes)** for OP_HASH160 / OP_SHA1, and **0x20 (32 bytes)** for OP_SHA256 / OP_HASH256 ? But here seems we have 0x20 instead of 0x14 for OP_HASH160.

Reply:
yes it will need to change, I am making tests for this right now

Reply:
As I can see it's related to 
```
OP_SIZE
OP_PUSHBYTES_1 0x20
OP_EQUALVERIFY
```
that checks that secret length is 32 bytes long. This check is to prevent attacks using very long secrets that might be supported by one coin, but not supported on another. It does not depend on hashing algorithm used.

Reply:
ah yes, reread what Decker was referring to, 0x20 is correct. 

https://gist.github.com/markblundeberg/7a932c98179de2190049f5823907c016 for some details as to why that is there

Reply:
Thanks for the link! I will take a final look at the PR tomorrow morning.

Reply:
> that checks that secret length is 32 bytes

aha, thanks for explanation. hard to check everything from mobile, that's why i wrote about doubts. but better to ask, than miss something and sorry.